### DISCUSSION TITLE
How to setup github authentication

### DISCUSSION THREAD

**Discussion by Khozhempo** [gogs/gogs#6965] [Q&A] [answered]
Hi!
Please, help. Docs do not help.
How to setup github authentication in gogs?

I create application in github and got Client ID. I added it to github.conf, which upload to /custom/conf/auth.d (and restart gogs).
In gogs "Edit Authentication Setting" set "API Endpoint" as "https://api.github.com"
In github OAuth application setting set URL for home page (as root for gogs site), but do not know what to set on "Authorization callback URL".
If I set "Authorization callback URL" same as home page, authorization on gogs do not work. Show incorrect user/psw.

Are any examples to setup this on gogs or some help to solve issue?
Thank you.

**Comment by unknwon** [comment, marked as answer]
Sorry about shitty docs at the moment 😭 

Gogs currently does not offer OAuth support, but the way to support GitHub authentication can be configured as:

Go to `/admin/auths/new` then

<img width="501" alt="CleanShot 2022-06-03 at 20 49 12@2x" src="https://user-images.githubusercontent.com/2946214/171857073-d9ba48bb-107d-405e-bb9d-7814933c38c2.png">

Users were able to use username+password but not anymore because GitHub.com has discounted support. To workaround, they'll need to use username + personal access token (as password field).

Though it definitely not optimal UX.
